### ISSUE TITLE
Accessibility issues with <Dropzone> ?

### ISSUE THREAD

**Issue by bradenmacdonald** [openedx/paragon#3957]
When using [the Dropzone component](https://paragon-openedx.netlify.app/components/dropzone/) in tests, its internal `<input type="file">` control does not appear to JSDom as an element with a `role`, and cannot be selected for testing using the `getByRole()` selector. I would expect it could be selected with something like `getByRole('button', { name: "Choose file" })`.

In Chrome, the internal `input` gets a `role=button` and implicit name `Choose File` supplied by the browser.

<img width="235" height="157" alt="Image" src="https://github.com/user-attachments/assets/b13ce363-adad-40d5-b59a-cebeebe56442" />

In Firefox, the internal `input` gets a `role=button` and implicit name `Browse… …` supplied by the browser.

<img width="1233" height="89" alt="Image" src="https://github.com/user-attachments/assets/65d06e7f-208f-4298-93c8-58063c1bc1f0" />

Should these be standardized by giving our internal input an explicit accessible name (e.g. `aria-label`) and `role=button` ? That would likely resolve the issue where JSdom isn't giving it a role, and allow us to use proper accessible queries in test code.

⚠️ Also, Firefix is giving a11y warnings on both the `input` ("Interactive elements must be focusable") and the parent `div` ("Focusable elements should have interactive semantics").
